jobs/integration/conftest.py

```diff
@@ -356,35 +344,36 @@ async def k8s_model(k8s_cloud, tools):
             "-c",
             tools.controller_name,
             tools.k8s_model_name,
-            tools.k8s_cloud,
+            k8s_cloud,
             "--config",
             "test-mode=true",
             "--no-switch",
         )
 
-        k8s_model = Model()
-        await k8s_model.connect(tools.k8s_connection)
-        yield k8s_model
+        _model_created = Model()
+        await _model_created.connect(tools.k8s_connection)
+        yield _model_created
     finally:
-        if not k8s_model:
-            return
-        await tools.run("juju-crashdump", "-a", "config", "-m", tools.k8s_connection)
-        click.echo("Cleaning up k8s model")
-        try:
-            for relation in k8s_model.relations:
+        if _model_created:
```

Comment:
we're careful here to not `return` in the finally, so exceptions can freely bubble out